src/OpenTelemetry.Collector.Dependencies/Implementation/HttpDesktopDiagnosticListener.net461.cs

```diff
@@ -0,0 +1,155 @@
+        {
+            this.options = options;
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            const string EventNameSuffix = ".OnStartActivity";
+            if (!(this.startRequestFetcher.Fetch(payload) is HttpWebRequest request))
+            {
+                CollectorEventSource.Log.NullPayload(nameof(HttpDesktopDiagnosticListener) + EventNameSuffix);
+                return;
+            }
+
+            this.Tracer.StartActiveSpanFromActivity(request.RequestUri.AbsolutePath, activity, SpanKind.Client, out var span);
+
+            if (span.IsRecording)
+            {
+                span.PutComponentAttribute("http");
+                span.PutHttpMethodAttribute(request.Method.ToString());
+                span.PutHttpHostAttribute(request.RequestUri.Host, request.RequestUri.Port);
+                span.PutHttpRawUrlAttribute(request.RequestUri.OriginalString);
+
+                if (this.options.SetHttpFlavor)
+                {
+                    span.PutHttpFlavorAttribute(request.ProtocolVersion.ToString());
+                }
+            }
+
+            if (!(this.options.TextFormat is TraceContextFormat))
+            {
+                this.options.TextFormat.Inject<HttpWebRequest>(span.Context, request, (r, k, v) => r.Headers.Add(k, v));
+            }
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            const string EventNameSuffix = ".OnStopActivity";
+            var span = this.Tracer.CurrentSpan;
+
+            if (span == null || !span.Context.IsValid)
+            {
+                CollectorEventSource.Log.NullOrBlankSpan(nameof(HttpDesktopDiagnosticListener) + EventNameSuffix);
+                return;
+            }
+
+            if (span.IsRecording)
+            {
+                if (this.stopResponseFetcher.Fetch(payload) is HttpWebResponse response)
+                {
+                    span.PutHttpStatusCode((int)response.StatusCode, response.StatusDescription);
+                }
+            }
+
+            span.End();
+        }
+
+        public override void OnException(Activity activity, object payload)
+        {
+            const string EventNameSuffix = ".OnException";
+            var span = this.Tracer.CurrentSpan;
+
+            if (span == null || !span.Context.IsValid)
+            {
+                CollectorEventSource.Log.NullOrBlankSpan(nameof(HttpDesktopDiagnosticListener) + EventNameSuffix);
+                return;
+            }
+
+            if (span.IsRecording)
+            {
+                if (!(this.stopExceptionFetcher.Fetch(payload) is Exception exc))
+                {
+                    CollectorEventSource.Log.NullPayload(nameof(HttpDesktopDiagnosticListener) + EventNameSuffix);
+                    return;
+                }
+
+                ProcessException(span, exc);
+            }
+
+            span.End();
```

Comment:
same as in OnStop. It will be great to make sure span will be ended even if callback failed with exception

Reply:
I added on all the listeners in the dependencies lib. Doing that I found a bug in SqlClient listener, it didn't have span.End call in exception event so I fixed that up. That pattern is used throughout the OT solution so probably need to update the other libs too.